File: app/src/test/java/com/benoitquenaudon/tvfoot/red/app/injection/component/TestComponent.java
package com.benoitquenaudon.tvfoot.red.app.injection.component;

import com.benoitquenaudon.tvfoot.red.app.domain.match.state.MatchStateBinderTest;
import dagger.Component;
import com.benoitquenaudon.tvfoot.red.app.common.schedulers.BaseSchedulerProvider;
import com.benoitquenaudon.tvfoot.red.app.domain.match.state.MatchRepository;
import com.benoitquenaudon.tvfoot.red.app.domain.matches.state.MatchesService;
import com.benoitquenaudon.tvfoot.red.app.injection.module.NetworkModule;
import com.benoitquenaudon.tvfoot.red.app.injection.module.ServiceMockModule;
import com.benoitquenaudon.tvfoot.red.app.injection.module.TestSchedulerModule;
import com.benoitquenaudon.tvfoot.red.testutil.Fixture;
import javax.inject.Singleton;
import org.jetbrains.annotations.NotNull;

@Singleton @Component(modules = {
    NetworkModule.class, //
    ServiceMockModule.class, //
    TestSchedulerModule.class, //
}) public interface TestComponent {
  Fixture fixture();

  BaseSchedulerProvider schedulerProvider();

  MatchesService matchesRepository();

  MatchRepository matchRepository();

  void inject(@NotNull MatchStateBinderTest matchStateBinderTest);
}

File: app/src/test/java/com/benoitquenaudon/tvfoot/red/injection/component/TestComponent.kt
package com.benoitquenaudon.tvfoot.red.injection.component

import com.benoitquenaudon.tvfoot.red.app.data.source.FakeMatchRepository
import com.benoitquenaudon.tvfoot.red.app.data.source.FakeMatchesRepository
import com.benoitquenaudon.tvfoot.red.app.data.source.FakePreferenceRepository
import com.benoitquenaudon.tvfoot.red.app.domain.match.state.MatchStateBinderTest
import com.benoitquenaudon.tvfoot.red.injection.module.FakeImplementationModule
import com.benoitquenaudon.tvfoot.red.injection.module.NetworkModule
import com.benoitquenaudon.tvfoot.red.injection.module.ServiceMockModule
import com.benoitquenaudon.tvfoot.red.injection.module.TestSchedulerModule
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = arrayOf(
    NetworkModule::class,
    ServiceMockModule::class,
    TestSchedulerModule::class,
    FakeImplementationModule::class))
interface TestComponent {
  fun inject(matchStateBinderTest: MatchStateBinderTest)
  fun inject(fakeMatchesRepository: FakeMatchesRepository)
  fun inject(fakeMatchRepository: FakeMatchRepository)
  fun inject(fakePreferenceRepository: FakePreferenceRepository)
}
